pyrefly/lib/lsp/wasm/hover.rs

```diff
@@ -175,17 +172,32 @@ impl HoverValue {
             .display
             .clone()
             .unwrap_or_else(|| self.type_.as_hover_string());
+        let snippet = format_hover_code_snippet(&self.type_, self.name.as_deref(), type_display);
```

Comment:
there should be a comment: why do we have to format_hover_code_snippet when we already have a type_display?

begs the question: can we instead make an `as_hover_string` that takes a fallback name? that would make this API one less step